src/webots/control/WbController.cpp

```diff
@@ -514,13 +514,7 @@ void WbController::setProcessEnvironment() {
       pythonSourceFile.close();
     }
 #ifdef __APPLE__
-    QProcess process;
-    process.setProcessEnvironment(env);
-    process.start(mPythonCommand, QStringList() << "-c"
-                                                << "import sys; print(sys.exec_prefix);");
-    process.waitForFinished();
-    const QString output = process.readAll();
-    if (output.startsWith("/usr/local/Cellar"))
+    if (mPythonCommand.startsWith("/usr/local/Cellar"))
```

Comment:
I fear that this might be a problem if users simply set `"python"` in `mPythonCommand` and modify their PATH environment variable prepending `/usr/local/Cellar/python@3.9/3.9.8/bin/` to find it there. Therefore, a better fix could be (not tested):
```suggestion
    QProcess process;
    process.setProcessEnvironment(env);
    process.start("which", QStringList() << mPythonCommand);
    process.waitForFinished();
    const QString output = process.readAll();
    if (output.startsWith("/usr/local/Cellar/python@"))
```
May I kindly ask you to test it?

Reply:
Sure!

Reply:
This work fine. I've updated my branch (squash + rebase onto master)